### PR TITLE
Upgrade demoshop to beta4 version

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Keep demoshop upgraded with newest version of Framework
|New feature| No 
|BC breaks| No 
|Fixes issues| -
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docker/conf/docker-compose-mac.yml.dist [hunk 1]
```diff
@@ -88,36 +75,27 @@ services:
...
             - "9200:9200"
         volumes:
-            - shopsys-framework-elasticsearch-data-sync:/usr/share/elasticsearch/data
+            - ./var/elasticsearch-data:/usr/share/elasticsearch/data:cached
         environment:
             - discovery.type=single-node
 
     microservice-product-search:
-        image: shopsys/microservice-product-search:v7.0.0-beta2
+        image: shopsys/microservice-product-search:latest
```
Comment (vitek-rostislav) [inline docker/conf/docker-compose-mac.yml.dist]: should not this be `v7.0.0-beta4`? I believe our upgrading notes are not mentioning this at all, and they link to master versions of the files, however, when upgrading, I think we should use the tagged versions (similarly to changes in `composer.json`)

File: tests/ShopBundle/Functional/Component/Doctrine/GroupedScalarHydratorTest.php [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 use Doctrine\ORM\Query\Expr\Join;
 use Shopsys\FrameworkBundle\Component\Doctrine\GroupedScalarHydrator;
 use Shopsys\FrameworkBundle\Model\Order\Item\OrderItem;
 use Shopsys\FrameworkBundle\Model\Order\Order;
-use Tests\ShopBundle\Test\DatabaseTestCase;
+use Tests\ShopBundle\Test\TransactionFunctionalTestCase;
 
-class GroupedScalarHydratorTest extends DatabaseTestCase
+class GroupedScalarHydratorTest extends TransactionFunctionalTestCase
```
Comment (vitek-rostislav) [inline tests/ShopBundle/Functional/Component/Doctrine/GroupedScalarHydratorTest.php:11]: it would be nice to mention in the commit message that `DatabaseTestCase` has been renamed to `TransactionFunctionalTestCase`

File: docker/conf/docker-sync.yml.dist [hunk 3]
```diff
@@ -3,4 +3,20 @@ syncs:
     shopsys-framework-sync:
         sync_userid: 501
         src: './'
-        sync_excludes: ['docker/']
+        sync_excludes: sync_excludes: [
```
Comment (vitek-rostislav) [inline docker/conf/docker-sync.yml.dist]: this is weird, you have the `sync_excludes` twice here, probably a typo?

File: app/config/paths.yml [hunk 4]
```diff
@@ -1,33 +1,34 @@
...
-    shopsys.front_design_image_url_prefix: /assets/frontend/images/
+    shopsys.front_design_image_url_prefix: '/assets/frontend/images/'
     shopsys.image_config_filepath: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/images.yml'
     shopsys.image_dir: '/web%shopsys.image_url_prefix%'
-    shopsys.image_url_prefix: /content/images/
+    shopsys.image_url_prefix: '/%shopsys.content_dir_name%/images/'
     shopsys.javascript_sources_dir: '%shopsys.resources_dir%/scripts'
-    shopsys.javascript_url_prefix: /assets/scripts/
-    shopsys.log_stream: /tmp/log-pipe
+    shopsys.javascript_url_prefix: '/assets/scripts/'
```
Comment (vitek-rostislav) [inline app/config/paths.yml:23]: Was that necessary to do? Moreover, strictly speaking, these changes should be done in a separate commit.

File: docker/php-fpm/Dockerfile [hunk 5]
```diff
@@ -58,21 +60,57 @@ ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
...
+
+########################################################################################################################
+
+FROM base as ci
+
+COPY --chown=www-data:www-data / /var/www/html
+
+RUN composer install --optimize-autoloader --no-interaction --no-progress
+
+RUN php phing composer-dev npm dirs-create test-dirs-create assets standards tests-static tests-acceptance-build
```
Comment (vitek-rostislav) [inline docker/php-fpm/Dockerfile:114]: The `test-dirs-create` should be added in https://github.com/shopsys/demoshop/pull/27/commits/c31f45ce777bd8a580538187111a5896b6a13647. The phing target does not exist at this moment, However, I see that it would be very difficult to update the Dockerfile gradually - you do not need to fix it here, we would just need to think about it when improving the upgrading process and instructions.

File: composer.json [hunk 6]
```diff
@@ -96,7 +96,7 @@
         "codeception/codeception": "^2.5.0",
         "phpstan/phpstan": "^0.7",
         "phpunit/phpunit": "^7.0",
-        "shopsys/coding-standards": "v7.0.0-beta2",
+        "shopsys/coding-standards": "v7.0.0-beta3",
```
Comment (vitek-rostislav) [inline composer.json]: It seems to me like a good idea to solve the standards first :+1:

Comment (vitek-rostislav) [reply composer.json]: I would just slightly edit the 2nd line of the commit message:

"- added missing @param and @return annotations through the whole codebase"

File: composer.json [hunk 7]
```diff
@@ -68,15 +68,15 @@
         "sensio/framework-extra-bundle": "^3.0.29",
         "sensio/generator-bundle": "^3.1.7",
         "shopsys/doctrine-orm": "2.6.2",
-        "shopsys/migrations": "v7.0.0-beta2",
-        "shopsys/form-types-bundle": "v7.0.0-beta2",
-        "shopsys/framework": "v7.0.0-beta2",
-        "shopsys/plugin-interface": "v7.0.0-beta2",
+        "shopsys/migrations": "v7.0.0-beta3",
```
Comment (vitek-rostislav) [inline composer.json]: "deps" - please, try not to use abbreviations in the commit messages :slightly_smiling_face:

Comment (vitek-rostislav) [reply composer.json]: I would say "dependencies on shopsys packages updated to beta3 version" to be more descriptive

File: build.xml [hunk 8]
```diff
@@ -271,17 +271,6 @@
         </exec>
     </target>
 
-    <target name="domains-urls-check" description="Checks the domain urls configuration.">
```
Comment (vitek-rostislav) [inline build.xml:274]: I am thinking that it would be probably nice to link the original pull requests on the 2nd line of your commit message :slightly_smiling_face: 

"- see https://github.com/shopsys/shopsys/pull/540"

File: app/config/packages/twig.yml [hunk 9]
```diff
@@ -7,7 +7,9 @@ twig:
         - '@ShopsysFramework/Admin/Form/fileuploadFields.html.twig'
         - '@ShopsysFramework/Admin/Form/imageuploadFields.html.twig'
         - '@ShopsysFramework/Admin/Form/pricetableFields.html.twig'
+        - '@ShopsysFramework/Admin/Form/orderItems.html.twig'
```
Comment (vitek-rostislav) [inline app/config/packages/twig.yml:10]: again, link to the original PR would be probably helpful when someone was trying to understand this change

File: build-dev.xml [hunk 10]
```diff
@@ -16,7 +16,7 @@
     <target name="standards" depends="phplint,ecs,phpstan,twig-lint,yaml-lint,eslint-check" description="Checks coding standards."/>
```
Comment (vitek-rostislav) [inline build-dev.xml:16]: you need to be more descriptive in commit message "phing target: Make couple of changes" - I do not know what happened and why :wink:

File: tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php [hunk 11]
```diff
@@ -4,58 +4,51 @@
...
-        $productData1 = $productDataFactory->create();
-        $productData1->name = [];
-        $productData1->price = 100;
-        $productData1->vat = $vat;
-        $productData1->priceCalculationType = Product::PRICE_CALCULATION_TYPE_AUTO;
-        $productMock = Product::create($productData1);
+        /** @var \Shopsys\ShopBundle\Model\Product\Product $product */
+        $product = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '1');
 
+        /* @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceCalculationForUser $productPriceCalculationForUser */
```
Comment (vitek-rostislav) [inline tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php:32]: should be "/** @var..." :slightly_smiling_face:

Comment (vitek-rostislav) [reply tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php:32]: however, this is inconsistent throughout multiple classes here, so I would not bother :wink:

File: tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php [hunk 12]
```diff
@@ -14,20 +13,17 @@
...
-use Shopsys\ShopBundle\Model\Product\Product;
-use Shopsys\ShopBundle\Model\Product\ProductDataFactory;
 use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
 use Tests\ShopBundle\Test\TransactionFunctionalTestCase;
 
 class InputPriceRecalculationSchedulerTest extends TransactionFunctionalTestCase
 {
     public function testOnKernelResponseNoAction()
     {
+        /** @var \Shopsys\FrameworkBundle\Component\Setting\Setting $setting */
```
Comment (vitek-rostislav) [inline tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php:25]: I wonder, how come that in some cases, you fixed these annotations? :laughing:

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi @grossmannmartin, good job :+1: I have added a few notes so please check them out :slightly_smiling_face:

As I mentioned in some of the notes, I think it would be nice to link the original pull requests in the commit messages where it makes sense. I could help with understanding the changes in the project.

File: docker/conf/docker-compose-mac.yml.dist (hunk 1 above)
Comment (grossmannmartin) [reply docker/conf/docker-compose-mac.yml.dist]: Yes, you're right. Fixed!

File: docker/conf/docker-sync.yml.dist (hunk 3 above)
Comment (grossmannmartin) [reply docker/conf/docker-sync.yml.dist]: Yes, it's a typo. Fixed!

File: app/config/paths.yml (hunk 4 above)
Comment (grossmannmartin) [reply app/config/paths.yml:23]: @vitek-rostislav Not exactly necessary. When I've changed `/content/` to a parameter, I had to add quotes to these values. 
And I end up unifying all the values. You're right, that not affected lines could be in a separate commit for clarity.

File: docker/php-fpm/Dockerfile (hunk 5 above)
Comment (grossmannmartin) [reply docker/php-fpm/Dockerfile:114]: I agree. Upgrade instructions can prevent this kind of mistakes.

File: tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php (hunk 11 above)
Comment (grossmannmartin) [reply tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php:32]: It would be nice to unify annotations. I've opened an issue to address this problem (#29)

File: tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php (hunk 12 above)
Comment (grossmannmartin) [reply tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php:25]: Yes, when I rewrote Tests, I tried to unify annotations, so it's easier to read a code. 
This should be unified (#29)

File: app/config/paths.yml (hunk 4 above)
Comment (grossmannmartin) [reply app/config/paths.yml:23]: Is it ok to leave it like this, or should I split it?

Comment (vitek-rostislav) [reply app/config/paths.yml:23]: Yes, it is ok to leave it as it is now...but next time... :slightly_smiling_face:

**Review by vitek-rostislav — APPROVED** [review]
Hi, the fixes are ok with me :+1: Let me know after the commit messages are improved :slightly_smiling_face:

**Review by vitek-rostislav — APPROVED** [review]
Hi @grossmannmartin, all the changes are now ok with me :+1: After the tests are done, you can move the task to business validation :tada:

File: build-dev.xml [hunk 13]
```diff
@@ -580,38 +580,11 @@
...
-    <target name="test-db-fixtures-demo-multidomain" description="Loads multidomain demo data fixtures excluding first domain into test DB.">
-        <if>
-            <istrue value="${is-multidomain}" />
-            <then>
-                <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
-                    <arg value="${path.bin-console}" />
-                    <arg value="--env=test" />
-                    <arg value="shopsys:fixtures:load" />
-                    <arg value="--fixtures=${path.framework}/src/DataFixtures/DemoMultidomain" />
-                    <arg value="--fixtures=${path.src}/Shopsys/ShopBundle/DataFixtures/DemoMultidomain" />
```
Comment (boris-brtan) [inline build-dev.xml:592]: hi, what do we want to do with upgrade of datafixtures ?

Comment (vitek-rostislav) [reply build-dev.xml:592]: oh yes :man_facepalming: Boris, thanks for noticing, we need to fix that :+1:

Comment (grossmannmartin) [reply build-dev.xml:592]: Data fixtures resolved in f86cf2dc2a028dbf2f5fa0e81ccec0954958c499

Comment (boris-brtan) [reply build-dev.xml:592]: OK, i see still 2 folders in Datafixtures but maybe it is ok for different project with different implementation as project-base

File: docker/conf/docker-compose-mac.yml.dist [hunk 14]
```diff
@@ -88,36 +75,27 @@ services:
...
+        image: shopsys/microservice-product-search:v7.0.0-beta4
         container_name: shopsys-framework-microservice-product-search
-        links:
-            - elasticsearch
-        depends_on:
-            - elasticsearch
 
     microservice-product-search-export:
-        image: shopsys/microservice-product-search-export:v7.0.0-beta2
+        image: shopsys/microservice-product-search-export:beta4
```
Comment (boris-brtan) [inline docker/conf/docker-compose-mac.yml.dist]: please fix

Comment (grossmannmartin) [reply docker/conf/docker-compose-mac.yml.dist]: Thanks for noticing. Fixed!
